### PR TITLE
ref(info): use conference existence as trigger for autoshowing dialog

### PR DESCRIPTION
**Pull request by virtuacoplenny**

### CODE REVIEW

**Comment by virtuacoplenny** [comment]
Rebased onto latest master.

**Comment by virtuacoplenny** [comment]
Jenkins test this please.

File: react/features/invite/components/InfoDialogButton.web.js [hunk 1]
```diff
@@ -136,15 +122,10 @@ class InfoDialogButton extends Component {
...
-    }
 
-    /**
-     * Clear the timeout to automatically show the {@code InfoDialog}.
-     *
-     * @inheritdoc
-     */
-    componentWillUnmount() {
-        clearTimeout(this._autoShowTimeout);
+        if (!this.props._isConferenceJoined && nextProps._isConferenceJoined) {
```
Comment (zbettenbuk) [inline react/features/invite/components/InfoDialogButton.web.js]: The function names (onDialogClose <-> maybeAutioShowDialog) seem to be opposites and this if seems to be disjunct with the previous one (including the conditions in the _maybeShowDialog function) so it's maybe worth merging them. Not a must have, it's just my OCD (either if or else :) )

File: react/features/invite/components/InfoDialogButton.web.js [hunk 2]
```diff
@@ -90,19 +92,7 @@ class InfoDialogButton extends Component {
     constructor(props) {
```
Comment (zbettenbuk) [inline react/features/invite/components/InfoDialogButton.web.js]: I see you nicely transformed this oldish class to a newer flow style, me likely! One more thing I always tell to Saul too is that I prefer using the inheritdoc annotation (without any other annotations) every time I don't intend to change the doc of the inherited field/function. Just if you wanna clean that up too.

Comment (virtuacoplenny) [reply react/features/invite/components/InfoDialogButton.web.js]: I used inheritdoc. I also rebased onto latest master (and force pushed).

File: react/features/invite/components/InfoDialogButton.web.js [hunk 3]
```diff
@@ -15,94 +16,82 @@ import { updateDialInNumbers } from '../actions';
...
+
+    /**
+     * Whether or not the toolbox, in which this component exists, is visible.
+     */
+    _toolboxVisible: boolean,
+
+    /**
+     * Invoked to toggle display of the info dialog.
+     */
+    dispatch: Function,
```
Comment (zbettenbuk) [inline react/features/invite/components/InfoDialogButton.web.js]: It's totally unrelated to this PR, just saw this line and made me wonder: Maybe we should use the Dispatch<*> type in the props too and that might help us avoid some flow type problems like we had yesterday? Don't change it, just to kick off this thinking

File: react/features/invite/components/InfoDialogButton.web.js [hunk 4]
```diff
@@ -15,94 +16,82 @@ import { updateDialInNumbers } from '../actions';
...
+     * Whether or not the local participant has joined a
+     * {@code JitsiConference}. Used to trigger auto showing of the
+     * {@code InfoDialog}.
+     */
+    _isConferenceJoined: Boolean,
+
+    /**
+     * The URL for a currently active live broadcast
+     */
+    _liveStreamViewURL: string,
```
Comment (zbettenbuk) [inline react/features/invite/components/InfoDialogButton.web.js]: I hate when other reviewers do this, but seems like I can't avoid it either. So I could have seen this earlier: can it be undefined? Maybe worth adding the ? to the type if yes.

**Pull request merged**